protostar/commands/test/test_result_structured_formatter.py

```diff
@@ -0,0 +1,193 @@
+from typing import Callable
+import json
+import math
+
+from protostar.protostar_exception import UNEXPECTED_PROTOSTAR_ERROR_MSG
+from protostar.testing import (
+    BrokenFuzzTestCaseResult,
+    BrokenTestCaseResult,
+    BrokenTestSuiteResult,
+    FailedFuzzTestCaseResult,
+    FailedTestCaseResult,
+    PassedFuzzTestCaseResult,
+    PassedTestCaseResult,
+    SkippedTestCaseResult,
+    TestResult,
+    UnexpectedBrokenTestSuiteResult,
+)
+from protostar.starknet.data_transformer import PythonData
+
+LogCallback = Callable[[str], None]
+
+
+# pylint: disable=too-many-return-statements
+def format_test_result_structured(test_result: TestResult) -> str:
+    if isinstance(test_result, PassedFuzzTestCaseResult):
+        return _format_passed_fuzz_test_case_result(test_result)
+    if isinstance(test_result, FailedFuzzTestCaseResult):
+        return _format_failed_fuzz_test_case_result(test_result)
+    if isinstance(test_result, BrokenFuzzTestCaseResult):
+        return _format_broken_fuzz_test_case_result(test_result)
+    if isinstance(test_result, PassedTestCaseResult):
+        return _format_passed_test_case_result(test_result)
+    if isinstance(test_result, FailedTestCaseResult):
+        return _format_failed_test_case_result(test_result)
+    if isinstance(test_result, BrokenTestCaseResult):
+        return _format_broken_test_case_result(test_result)
+    if isinstance(test_result, SkippedTestCaseResult):
+        return _format_skipped_test_case_result(test_result)
+    if isinstance(test_result, UnexpectedBrokenTestSuiteResult):
+        return _format_unexpected_exception_test_suite_result(test_result)
+    if isinstance(test_result, BrokenTestSuiteResult):
+        return _format_broken_test_suite_result(test_result)
+    raise NotImplementedError("Unreachable")
+
+
+def _format_passed_test_case_result(
+    passed_test_case_result: PassedTestCaseResult,
+) -> str:
+    return _format_passed_fuzz_test_case_result(
+        PassedFuzzTestCaseResult(
+            captured_stdout=passed_test_case_result.captured_stdout,
+            execution_resources=passed_test_case_result.execution_resources,
+            file_path=passed_test_case_result.file_path,
+            execution_time=passed_test_case_result.execution_time,
+            test_case_name=passed_test_case_result.test_case_name,
+            fuzz_runs_count=None,
+        )
+    )
+
+
+def _format_failed_test_case_result(
+    failed_test_case_result: FailedTestCaseResult,
+) -> str:
+    result: PythonData = {
```

Comment:
I don't see a conceptual connection between formatting result and `PythonData`. The result of formatting should be just a dict that can be represented as JSON.

Reply:
This is correct, I used it because it's essentially `Dict[str, Any]` which seemed to fit the purpose. I can just rename it for this case, no problem (I'm not sure if there's a way to type a JSON-serializable dict).